tools/fork-network/src/cli.rs

```diff
@@ -800,22 +801,39 @@ impl ForkNetworkCommand {
         new_validator_accounts: Vec<AccountInfo>,
         epoch_manager: Arc<EpochManagerHandle>,
         home_dir: &Path,
-        near_config: &NearConfig,
+        near_config: &mut NearConfig,
     ) -> anyhow::Result<()> {
-        let epoch_config = epoch_manager.get_epoch_config(epoch_id)?;
-        let protocol_version = match protocol_version {
+        // At this point, the genesis will contain the chain id of the original chain.
+        // We need to load the epoch config for the new chain id and new genesis protocol version.
+        let new_chain_id = chain_id
+            .clone()
+            .unwrap_or_else(|| near_config.genesis.config.chain_id.clone() + chain_id_suffix);
+        near_config.genesis.config.chain_id = new_chain_id.clone();
+
+        let genesis_protocol_version = match protocol_version {
             Some(v) => v,
-            None => {
-                let epoch_info = epoch_manager.get_epoch_info(epoch_id)?;
-                epoch_info.protocol_version()
-            }
+            None => near_config.genesis.config.protocol_version,
+        };
+        near_config.genesis.config.protocol_version = genesis_protocol_version;
+
+        // This is based on the assumption that epoch length is part of genesis config and not epoch config.
+        near_config.genesis.config.epoch_length = epoch_length;
+
+        let epoch_config_dir = home_dir.join("epoch_configs");
+        tracing::info!(target: "fork-network", "maybe loading epoch configs from {:?}", epoch_config_dir);
```

Comment:
can you move different versions of this to if and else branched (eg. "loading configs from ..." and "not loading...") so we can also see whether it happened or not?